File: Topological sort[01-06-20203]/Solution.cpp
//  >> C++ CODE <<

vector<int> topoSort(int V, vector<int> adj[]) {
    vector<int> indegree(V, 0);

    for (int i = 0; i < V; ++i) {
        for (int nn : adj[i]) {
            indegree[nn]++;
        }
    }

    queue<int> queue;
    vector<int> ans;

    for (int node = 0; node < V; ++node) {
        if (indegree[node] == 0) {
            queue.push(node);
        }
    }

    while (!queue.empty()) {
        int node = queue.front();
        queue.pop();
        ans.push_back(node);

        for (int i : adj[node]) {
            indegree[i]--;
            if (indegree[i] == 0) {
                queue.push(i);
            }
        }
    }

    return ans;
}

File: Topological sort[01-06-20203]/Solution.py
# >> PYTHON CODE <<

class Solution:
    
    #Function to return list containing vertices in Topological order.
    def topoSort(self, V, adj):
        indegree=[0]*V
        
        for i in range(V):
            for nn in adj[i]:
                indegree[nn]+=1

        queue=[]
        ans=[]
        
        for node in range(V):
            if(indegree[node]==0):
                queue.append(node)
        
        
        while(len(queue)>0):
            node=queue[0]
            queue.pop(0)
            ans.append(node)
            
            for i in adj[node]:
                indegree[i]-=1
                if((indegree[i]) == 0):
                    queue.append(i)
        
        return ans